src/iexamine.cpp

```diff
@@ -2780,6 +2783,29 @@ void iexamine::water_source(player &p, const tripoint &examp)
     g->handle_liquid( water, nullptr, 0, &examp );
 }
 
+void iexamine::milk_source( monster *source_mon )
+{
+    item milk( "milk", 0, 1 );
+
+    // Max duration set here to avoid including effect.h
+    int max_dur = 14400;
+
+    // If that takes care of checking if there's milk left
+    if( !source_mon->has_effect( effect_milked ) ) {
+        g->handle_liquid( milk, nullptr, 0, nullptr, nullptr, source_mon );
+        add_msg( _( "You milk the %s." ), source_mon->disp_name().c_str() );
+    } else if( source_mon->has_effect( effect_milked ) &&
```

Comment:
This `has_effect` check is redundant. The branch will only be taken when the monster has the effect, otherwise the first `if` condition already yields `true`.